validator/app/src/compute_horde_validator/validator/allowance/tests/mp_helpers.py

```diff
@@ -0,0 +1,131 @@
+"""Spawn-safe multiprocessing helpers for allowance concurrency tests.
+
+These functions avoid importing Django models at module import time so that
+`django.setup()` can be called first inside the child process when using the
+`spawn` start method.
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from multiprocessing import Queue
+from typing import Any
+
+import django
+from compute_horde_core.executor_class import ExecutorClass
+from django.apps import apps
+from django.conf import settings
+
+from ..default import allowance
+from ..types import CannotReserveAllowanceException, ReservationAlreadySpent, ReservationNotFound
+from .mockchain import set_block_number
+
+
+@dataclass
+class ReservationResult:
+    status: str  # "ok" or "fail"
+    reservation_id: int | None
+    data: Any | None  # list[int] for ok, float (available) for fail
+
+
+QueuePayload = ReservationResult
+
+
+def _django_bootstrap():
+    if not settings.configured:
+        django.setup()
+    else:
+        # Ensure app registry ready
+        if not apps.ready:
+            django.setup()
+
+
+def reserve_worker(
+    miner: str,
+    exec_class_value: str,
+    job_start_block: int,
+    amount: float,
+    q: Queue[QueuePayload],
```

Comment:
I think that `QueuePayload` alias is not needed. 
```suggestion
    q: Queue[ReservationResult],
```